I could only get the spack workflow to work if I turn aec default=False. Otherwise, I get the error below. 

```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j2' 'test'

3 warnings found in build log:
     25     -- Copying test file ref_gdaswave.t00z.wcoast.0p16.f000.grb2index2
     26     -- Copying test file g1.grib2
     27     -- Copying test file gdaswave.t00z.wcoast.0p16.f000.grib2
     28     -- Copying test file ref_test_jpcpack_4_output.txt
     29     -- Configuring done (0.4s)
     30     -- Generating done (0.1s)
  >> 31     CMake Warning:
     32       Manually-specified variables were not used by the project:
     33     
     34         CMAKE_POLICY_DEFAULT_CMP0090
     35         G2C_COMPARE
     36     
     37     

     ...

     120    
     121      678 |      call simunpack(cgrib(ipos), lensec-5, idrstmpl, ndpts,
             fld)
     122          |                                                            
            2
     123    ......
     124      686 |      call simunpack(cgrib(ipos), lensec-5, idrstmpl, ndpts-
            1, fld(2))
     125          |                                                            
              1
  >> 126    Warning: Element of assumed-shape or pointer array as actual argume
            nt at (1) cannot correspond to actual argument at (2)
     127    [ 19%] Building Fortran object src/CMakeFiles/g2_4_f.dir/g2create.F
            90.o
     128    cd /tmp/runner/spack-stage/spack-stage-g2-develop-w4caohczsnsc3axnf
            hyjqont476hjnpg/spack-build-w4caohc/src && /home/runner/work/NCEPLI
            BS-g2/NCEPLIBS-g2/spack-root/lib/spack/env/gcc/gfortran -DKIND=4 -D
            USE_AEC  -g -funroll-loops  -fallow-argument-mismatch -O3 -Jinclude
            _4 -c /home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/spack-devpkg-g2/src
            /g2create.F90 -o CMakeFiles/g2_4_f.dir/g2create.F90.o
     129    /home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/spack-devpkg-g2/src/g2cre
            ate.F90:446:18:
     130    
     131      442 |      call simpack(pfld, ndpts, idrstmpl, cpack, lcpack)
     132          |                  2
     133    ......
     134      446 |      call simpack(pfld(2), ndpts-1, idrstmpl, cpack, lcpack
            )
     135          |                  1
  >> 136    Warning: Element of assumed-shape or pointer array as actual argume
            nt at (1) cannot correspond to actual argument at (2)
     137    [ 20%] Building Fortran object src/CMakeFiles/g2_4_f.dir/g2sim.F90.
            o
     138    cd /tmp/runner/spack-stage/spack-stage-g2-develop-w4caohczsnsc3axnf
            hyjqont476hjnpg/spack-build-w4caohc/src && /home/runner/work/NCEPLI
            BS-g2/NCEPLIBS-g2/spack-root/lib/spack/env/gcc/gfortran -DKIND=4 -D
            USE_AEC  -g -funroll-loops  -fallow-argument-mismatch -O3 -Jinclude
            _4 -c /home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/spack-devpkg-g2/src
            /g2sim.F90 -o CMakeFiles/g2_4_f.dir/g2sim.F90.o
     139    /usr/local/bin/cmake -E touch src/CMakeFiles/g2_4_f.dir/drstemplate
            s.F90.o.provides.build
     140    /usr/local/bin/cmake -E touch src/CMakeFiles/g2_4_f.dir/g2grids.F90
            .o.provides.build
     141    /usr/local/bin/cmake -E touch src/CMakeFiles/g2_4_f.dir/g2logging.F
            90.o.provides.build
     142    /usr/local/bin/cmake -E touch src/CMakeFiles/g2_4_f.dir/gribmod.F90
            .o.provides.build

See build log for details:
  /tmp/runner/spack-stage/spack-stage-g2-develop-w4caohczsnsc3axnfhyjqont476hjnpg/spack-build-out.txt

See test log for details:
  /tmp/runner/spack-stage/spack-stage-g2-develop-w4caohczsnsc3axnfhyjqont476hjnpg/install-time-test-log.txt

Error: Process completed with exit code 1.
```

--- a/spack/package.py
+++ b/spack/package.py
@@ -37,7 +37,7 @@ class G2(CMakePackage):
         when="@3.4.6:",
     )
     variant("w3emc", default=True, description="Enable GRIB1 through w3emc", when="@3.4.6:")
-    variant("aec", default=False, description="Use AEC library", when="@develop")
+    variant("aec", default=True, description="Use AEC library", when="@develop")
     variant("shared", default="False", when="@3.4.7:")
     variant("openmp", default=False, description="Use OpenMP multithreading")
     variant("utils", default=False, description="Build grib utilities")

--- a/spack/package.py
+++ b/spack/package.py
@@ -46,7 +46,7 @@ class G2(CMakePackage):
     depends_on("jasper@:2.0.32", when="@:3.4.7")
     depends_on("jasper")
     depends_on("g2c", when="@develop")
-    depends_on("g2c +aec", when="+aec")
+    depends_on("g2c@develop +aec", when="+aec")
     depends_on("libpng")
     depends_on("zlib-api")
     depends_on("bacio", when="@3.4.6:")

